File: formularios/7. sueldo/js/sueldo.js
/**
 * Calcular el sueldo total de una persona
 * Autor: Santiago GM
 * 02/04/2024
 */

function datos() {

  let diaT =  parseInt(document.getElementById('diaT').value)
  let valorD =  parseFloat(document.getElementById('valorD').value)
  let pantalla = document.getElementById('pantalla')

  
  if (diaT && valorD) {
    
    pantalla.innerHTML = 
    `
      <h5>Calcular el sueldo total de una persona: </h5>
      <br>
      <p>
      
      Su salario Bruto es: $${sueldo(diaT,valorD)} <br>
      Sus deducibles son: <br>
      El porcentaje salud es: ${salud(sueldo(diaT,valorD))}% <br>
      El porcentaje de pensión es: ${pension(sueldo(diaT,valorD))}% <br>
      El porcentaje de ARL es: ${arl(sueldo(diaT,valorD))}% <br>
      El pago Neto será de: $${pagoTotal(sueldo(diaT,valorD))}
      
      </p>
    `
  } else {
        pantalla.innerHTML = `<strong>Todos los campos deben esta llenos </strong>`
    
  }

    return false
}

function sueldo(diaT, valorD) {
    let pago = diaT * valorD;
    return pago;
  }
  
  function salud(pago) {
    let saludP = pago * 0.12;
    return saludP;
  }
  
  function pension(pago) {
    let pensionP = pago * 0.16;
    return pensionP;
  }
  
  function arl(pago) {
    let arlP = pago * 0.052;
    return arlP;
  }
  function pagoTotal(pago) {
  
    let saludP = salud(pago);
    let pensionP = pension(pago);
    let arlP = arl(pago);
    let total = pago - (saludP + pensionP + arlP);
    return total;
  }
  

File: python/7. sueldo/utils/sueldo.py

# Calcular el sueldo total de una persona
# Autor: Santiago GM

# Como parámetro
def sueldo(dias_trabajados_p, valor_dia_p) :
  dias_trabajados, valor_dia = dias_trabajados_p, valor_dia_p
  pago = dias_trabajados * valor_dia
  return pago

def salud(pago):
  salud_p = pago * 0.12
  return salud_p

def pension(pago):
  pension_p = pago * 0.16
  return pension_p


def arl(pago):
  arl_p = pago * 0.052
  return arl_p

def pago_total(pago):
  salud_p = salud(pago)
  pension_p = pension(pago)
  arl_p = arl(pago)
  total = pago - (salud_p + pension_p + arl_p)
  return total


# como función lambda

sueldo_lamb =  lambda dias_trabajados_p, valor_dia_p: dias_trabajados_p * valor_dia_p
salud_lamb = lambda pago: pago * 0.12
pension_lamb =  lambda pago: pago * 0.16
arl_lamb =  lambda pago: pago * 0.052
pago_Total_lamb =  lambda pago: pago - (salud_lamb(pago) + pension_lamb(pago) + arl_lamb(pago))


